components/humantask/org.wso2.carbon.humantask/src/main/java/org/wso2/carbon/humantask/core/engine/runtime/xpath/JaxpFunctionResolver.java

```diff
@@ -303,31 +392,32 @@ public Object evaluate(List args) throws XPathFunctionException {
             }
             return createOrgEntity(excludedOwners);
         }
+
+        private GenericHumanRoleDAO getExcludedOwnersFromCtx() {
+            return evalCtx.getGenericHumanRole(GenericHumanRoleDAO.GenericHumanRoleType.EXCLUDED_OWNERS);
+        }
     }
 
     /**
      * Returns the initiator of the task. It MUST evaluate to an empty htt:user in case there is no initiator.
      * <p/>
      * If the task name is not present the current task MUST be considered.
+     * In : task name (optional)
+     * Out :the task initiator (user id as htt:user)
      */
     public class GetTaskInitiator implements XPathFunction {
+
         public Object evaluate(List args) throws XPathFunctionException {
-            GenericHumanRoleDAO taskInitiator = null;
             String username = "";
-            if (args.size() == 0) {
-                // Case 1: consider current Task.
-                taskInitiator = evalCtx.getGenericHumanRole(
-                        GenericHumanRoleDAO.GenericHumanRoleType.TASK_INITIATOR);
-                for (OrganizationalEntityDAO oe : taskInitiator.getOrgEntities()) {
-                    if (OrganizationalEntityDAO.OrganizationalEntityType.USER.equals(oe.getOrgEntityType())) {
-                        username = oe.getName();
-                        break;
-                    }
-                }
-            } else if ((args.size() == 1) && (args.get(0) instanceof String)) {
+            if (args.size() == 0)
```

Comment:
Fixed it in https://github.com/wso2/carbon-business-process/pull/227. Thank you.